lib/rsync_store.ml

```diff
@@ -125,7 +125,9 @@ let cache ~user t name =
   (* Create writeable clone. *)
   let gen = cache.gen in
   let { Obuilder_spec.uid; gid } = user in
-  Rsync.copy_children ~chown:(Printf.sprintf "%d:%d" uid gid) ~src:snapshot ~dst:tmp () >>= fun () ->
+  (* rsync --chown not supported by the rsync that macOS ships with *)
```

Comment:
Shouldn’t `rsync` be using `-a` (which includes `-o` and `-g`, aka. preserve owner/group) instead of plain rsync + chown? It should be more efficient.

Or are there some issues with owner/group needing changing between 2 snapshots?

Reply:
By default all `rsync` commands have `-aHq`

https://github.com/ocurrent/obuilder/blob/ea78b3544c11d8830f7d236b360728c9be8c2fa8/lib/rsync_store.ml#L27

The extra `chown` iiuc is for differences between uids/gids when restoring caches. It is not used in macOS where we ignore the `config.user` because we're using `users` to build, unlike in runc where it could be anyone 

https://github.com/ocurrent/obuilder/blob/ea78b3544c11d8830f7d236b360728c9be8c2fa8/lib/runc_sandbox.ml#L110-L115